cmd/boulder/main_test.go

```diff
@@ -0,0 +1,74 @@
+package main
+
+import (
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/letsencrypt/boulder/cmd"
+	"github.com/letsencrypt/boulder/test"
+)
+
+// TestConfigValidation checks that each of the components which register a
+// validation tagged Config struct at init time can be used to successfully
+// validate their corresponding test configuration files.
+func TestConfigValidation(t *testing.T) {
+	configPath := "../../test/config"
+	if os.Getenv("BOULDER_CONFIG_DIR") == "test/config-next" {
+		configPath = "../../test/config-next"
+	}
+
+	// Each component is a set of `cmd` package name and a list of paths to
+	// configuration files to validate.
+	components := make(map[string][]string)
+
+	// For each component, add the paths to the configuration files to validate.
+	// By default we assume that the configuration file is named after the
+	// component. However, there are some exceptions to this rule. We've added
+	// special cases for these components.
+	for _, cmdName := range cmd.AvailableConfigs() {
+		var fileNames []string
+		switch cmdName {
```

Comment:
nit: the cases in this switch/case are not in any perceivable order. Alphabetical order may be best?